bika/lims/idserver.py

```diff
@@ -55,14 +55,23 @@ def get_contained_items(obj, spec):
     return obj.objectItems(spec)
 
 
+def get_type_id(context, **kw):
+    """Returns the type id for the context passed in
+    """
+    portal_type = kw.get("portal_type") or api.get_portal_type(context)
+    if IAnalysisRequestPartition.providedBy(context):
+        portal_type = "{}Partition".format(portal_type)
```

Comment:
Allow overriding from the passed in keywords also for Partitions.
The precedence would be then

1. Portal Type
2. Interface Lookup
3. Keyword override

Reply:
Not sure if I understand correctly. You mean to override `kw`'s value for "portal_type" when the context is a `IAnalysisRequestPartition`?

Reply:
Yes exactly. If the user provided `portal_type=foo`, it should overrule the `portal_type` and even the fact that `IAnalysisRequestPartition` is provided by the context